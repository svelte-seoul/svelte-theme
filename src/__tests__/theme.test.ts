import {render} from "@testing-library/svelte";

import SampleComponent from "../../test/SampleComponent.svelte";

describe('svelte-theme', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  
  it("renders without error", () => {
    const testngLib = render(SampleComponent);
  
    expect(testngLib.container).toMatchSnapshot();
  });
});
