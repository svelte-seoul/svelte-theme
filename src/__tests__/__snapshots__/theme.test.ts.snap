// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svelte-theme renders without error 1`] = `
<body>
  <div>
    <div />
  </div>
</body>
`;
